### PR TITLE
Submit coverage reports from AppVeyor

### LINKED ISSUE #124: Support submit from Appveyor
The [Codecov submit function](https://github.com/JuliaCI/Coverage.jl/blob/e8008cc90f61a48ef293cbf6e6a822bd532aabf9/src/codecovio.jl#L67) only works from within Travis CI. I found the note below while getting the Repository Upload Token on Codecov:

> Note: Token not required for public repositories uploading from Travis, CircleCI or AppVeyor.

From this comment and from inspecting the codecov-bash source it should definitely be possible to support Appveyor. The relevant Appveyor environmental variables can be found [here](https://github.com/codecov/codecov-bash/blob/master/codecov#L579-L585)

### PR DESCRIPTION
**Pull request by christopher-dG**
These changes let you push coverage reports to CodeCov from AppVeyor builds. However, for now this will only be useful for Julia versions <= 0.4 (see [here](https://github.com/JuliaLang/julia/issues/21289)).

Closes #124

### CODE REVIEW

**Comment by christopher-dG** [comment]
Hm, I'll work on getting the tests passing tomorrow.

**Comment by christopher-dG** [comment]
Added Coveralls support too.

File: README.md [hunk 1]
```diff
@@ -114,11 +120,17 @@ REPO_TOKEN=$YOUR_TOKEN_HERE julia -e 'cd(Pkg.dir("MyPkg")); using Coverage; Code
 3. Use the command line option when you run your tests
   * Either with something like `julia --code-coverage test/runtests.jl`, or
   * with something like  `julia -e 'Pkg.test("MyPkg", coverage=true)'`
-4. Add the following to the end of your `.travis.yml` or `appveyor.yml` file. This line downloads this package, collects the per-file coverage data, then bundles it up and submits to Coveralls. Coverage.jl assumes that the working directory is the package directory, so it changes to that first (so don't forget to replace `MyPkg` with your package's name!
-```yml
-after_success:
-- julia -e 'cd(Pkg.dir("MyPkg")); Pkg.add("Coverage"); using Coverage; Coveralls.submit(process_folder())'
-```
+4. Add the following to the end of your `.travis.yml` or `.appveyor.yml` file. This line downloads this package, collects the per-file coverage data, then bundles it up and submits to Codecov. Coverage.jl assumes that the working directory is the package directory, so it changes to that first (so don't forget to replace `MyPkg` with your package's name!
```
Comment (tkelman) [inline README.md]: "submits to Coveralls" for this one?

Comment (christopher-dG) [inline README.md]: Fixed, thank you.

File: src/codecovio.jl [hunk 2]
```diff
@@ -62,18 +62,42 @@ module Codecov
...
-        kwargs = set_defaults(kwargs,
-            service      = "travis-org",
-            branch       = ENV["TRAVIS_BRANCH"],
-            commit       = ENV["TRAVIS_COMMIT"],
-            pull_request = ENV["TRAVIS_PULL_REQUEST"],
-            job          = ENV["TRAVIS_JOB_ID"],
-            slug         = ENV["TRAVIS_REPO_SLUG"],
-            build        = ENV["TRAVIS_JOB_NUMBER"],
+        if haskey(ENV, "APPVEYOR") && lowercase(ENV["APPVEYOR"]) == "true"
+            appveyor_pr = haskey(ENV, "APPVEYOR_PULL_REQUEST_NUMBER") ?
```
Comment (tkelman) [inline src/codecovio.jl]: this kind of pattern of `haskey ? getenv : ...` would probably be cleaner with the 3-arg form `get(ENV, key, val_if_not_present)`

**Comment by tkelman** [comment]
This looks really nicely done, thanks! What happens if the same repo is sending coveralls / codecov reports from multiple CI services?

File: src/codecovio.jl (hunk 2 above)
Comment (christopher-dG) [inline src/codecovio.jl]: Ah I forgot about that, it makes perfect sense for the AppVeyor PR number.

For the `ENV["TRAVIS_CI"/"APPVEYOR"]` ones, how does this look?

```if lowercase(get(ENV, $the_ci_name", "false")) == "true"```

**Comment by christopher-dG** [comment]
I think the coverage values are combined, i.e. if a Travis build covers one half of the code and an AppVeyor build covers the other half, then coverage is considered 100% (which makes sense if your code paths depend on OS). I'll run a test on that to confirm though.

**Comment by christopher-dG** [comment]
It looks like Codecov properly combines multiple builds, Coveralls only takes the latest build.

File: src/codecovio.jl [hunk 3]
```diff
@@ -85,7 +84,7 @@ module Codecov
                 slug         = ENV["APPVEYOR_REPO_NAME"],
                 build        = ENV["APPVEYOR_JOB_ID"],
             )
-        elseif haskey(ENV, "TRAVIS_CI") && lowercase(ENV["TRAVIS_CI"]) == "true"
+        elseif haskey(ENV, "TRAVIS") && lowercase(ENV["TRAVIS"]) == "true"
```
Comment (tkelman) [inline src/codecovio.jl]: get here too?

Comment (christopher-dG) [inline src/codecovio.jl]: Oops, fixed.

**Comment by musm** [comment]
bump,?

**Pull request merged**